buildSrc/src/main/java/org/elasticsearch/gradle/ReaperExtension.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.gradle;
+
+import org.gradle.api.provider.Provider;
+
+public class ReaperExtension {
```

Comment:
This is an interesting pattern. For the docker stuff I actually added a utility for looking up global services services by type:
https://github.com/elastic/elasticsearch/blob/bae65dde3ff788673fb9d3607486ba1bdce07d3d/buildSrc/src/main/java/org/elasticsearch/gradle/DistributionDownloadPlugin.java#L54-L57

I'm wondering which we should use? I don't have strong feelings either way but we should probably make them both consistent.

Reply:
Interesting. Haven't' seen that. I'm all in for consistency here.  I'm moving this PR to use your pattern. My reasoning:

- Having an extension just to carry a property around seems a bit wrong to me. Your approach is more direct without polluting the area of surface within the DSL (Meaning we should avoid where possible to introduce extensions where the user cannot or is not intended to do anything with them)
 
Though there's some more risk to hit internal api changes with the dockerSupport example I guess. But let's see. We can always move back if we hit a barrier.

Reply:
> Though there's some more risk to hit internal api changes with the dockerSupport example I guess. But let's see. We can always move back if we hit a barrier.

Indeed. The APIs seems to be missing a convenient way to subsequently retrieve these global services. Ideally the would work with `@Inject`. We might want to chat with the build tool folks about what is the idiomatic way to use these things.